data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/BinlogClientWrapper.java

```diff
@@ -54,4 +68,27 @@ public void disconnect() throws IOException {
     public BinaryLogClient getBinlogClient() {
         return binlogClient;
     }
+
+    private void categorizeError(Exception e) {
+        if (e instanceof AuthenticationException) {
+            rdsSourceAggregateMetrics.getStream4xxErrors().increment();
+            rdsSourceAggregateMetrics.getStreamAuthErrors().increment();
+            LOG.error("Failed to connect to replication stream: Authentication failed. [{}]", e.getMessage());
+        } else if (e.getCause() != null && e.getCause().getMessage() != null && e.getCause().getMessage().contains(CONNECTION_REFUSED)) {
+            rdsSourceAggregateMetrics.getStream4xxErrors().increment();
+            rdsSourceAggregateMetrics.getStreamServerNotFoundErrors().increment();
+            LOG.error("Failed to connect to replication stream: Cannot connect to MySQL server. [{}]", e.getMessage());
+        } else if (e.getMessage() != null && e.getMessage().contains(FAILED_TO_DETERMINE_BINLOG_FILENAME)) {
```

Comment:
Is it possible to check through exception type?

Reply:
They are mostly just generic IOException so I didn't use exception type except the `AuthenticationException`